src/sentry/tasks/auth.py

```diff
@@ -112,3 +139,64 @@ def _remove_2fa_non_compliant_member(member, org, actor=None, actor_key=None, ip
             context=email_context,
         )
         message.send_async([member.email])
+
+
+@instrumented_task(
+    name="sentry.tasks.remove_2fa_non_compliant_members",
+    queue="auth",
+    default_retry_delay=60 * 5,
+    max_retries=5,
+)
+def remove_2fa_non_compliant_members(org_id, actor_id=None, actor_key_id=None, ip_address=None):
+    TwoFactorComplianceTask().remove_non_compliant_members(
+        org_id, actor_id, actor_key_id, ip_address
+    )
+
+
+class VerifiedEmailComplianceTask(OrganizationComplianceTask):
+    log_label = "verified email"
+
+    def is_compliant(self, member: OrganizationMember) -> bool:
+        return UserEmail.get_primary_email(member.user).is_verified
+
+    def call_to_action(self, member: OrganizationMember):
+        user = member.user
+        if isinstance(user, User):
```

Comment:
it seems that this check will result in an email not being sent initially to the user when they are kicked out, at least locally when i'm running it -- we remove the user from the member before calling `call_to_action`, so when it gets to here user is `None`. maybe `call_to_action` could optionally take in a user param?

Reply:
Let me see if I can repro the `None` case. I believe I added the `isinstance` check to guard against an edge case where `user` is a `django.contrib.auth.models.User`, because `UserEmail.get_primary_email` will choke unless it's a `sentry.models.user.User`.

Reply:
Oof, I think I see the problem -- I fixed this https://github.com/getsentry/sentry/pull/25972/commits/eab3d34ae93f7b32d045d9b83ed7dc4b13cff0e4 in the wrong direction. That's a heck of a silent failure; thanks @JoshFerge for the catch.

I'll tighten up the check and add a warning in the expected `django.contrib.auth.models.User` case.